tools/topology/topology2/cavs/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Array of "input-file-name;output-file-name;comma separated pre-processor variables"
+set(TPLGS
+# CAVS HDMI only topology with passthrough pipelines
+"cavs-passthrough-hdmi\;cavs-passthrough-hdmi"
+# CAVS HDA topology with passthrough pipelines
+"cavs-passthrough-hdmi\;cavs-passthrough-hda\;HDA_CONFIG=passthrough"
+# CAVS HDMI only topology with gain-based pipelines
+"cavs-gain-hdmi\;cavs-gain-hdmi"
+# CAVS HDA topology with gain-based pipelines
+"cavs-gain-hdmi\;cavs-gain-hda\;HDA_CONFIG=gain"
+# CAVS HDA topology with mixer-based pipelines
+"cavs-passthrough-hdmi\;cavs-mixin-mixout-hda\;HDA_CONFIG=mix"
+)
+
+# generate ABI
+execute_process(
+	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/get_abi.sh ${SOF_ROOT_SOURCE_DIRECTORY}
+	OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/abi.conf
+)
+
+add_custom_target(topology2_cavs ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory . ${SOF_TOPOLOGY_BINARY_DIRECTORY}
+)
+
+add_dependencies(topologies2 topology2_cavs)
+
+foreach(tplg ${TPLGS})
+	set(defines "")
+	list(LENGTH tplg length)
+	list(GET tplg 0 input)
+	list(GET tplg 1 output)
+
+	math(EXPR last_index "${length}-1")
+
+	if (${last_index} EQUAL 2)
+		list(GET tplg ${last_index} defines)
+	endif()
+
+	# copy ABI and input conf file contents
+	configure_file(${CMAKE_CURRENT_BINARY_DIR}/abi.conf ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf)
+	file(READ ${CMAKE_CURRENT_SOURCE_DIR}/${input}.conf CONTENTS)
+	file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf "${CONTENTS}")
+
+# Note: this does NOT use VERBATIM, see explanation in ../topology/CMakeLists.txt
+	add_custom_command(
+		OUTPUT ${output}.tplg
+		# -p to pre-process Topology2.0 conf file
+		COMMAND alsatplg \$\${VERBOSE:+-v 1} -I ${CMAKE_CURRENT_SOURCE_DIR} -D ${defines} -p -c ${output}.conf -o ${output}.tplg
```

Comment:
Are more subdirectories besides `cavs/` expected and if yes will they use the same `alsatplg` command? If yes and yes please make this a new `add_alsatplg2_command()` macro next to the existing `add_alsatplg_command()`